### PR TITLE
Ok/abstracting business logic

### PR DESCRIPTION
**Pull request by dacoaster**

### CODE REVIEW

File: src/modules/FileSystemUtility.js [hunk 1]
```diff
@@ -61,7 +61,8 @@ module.exports.exportItems = async (ids) => {
 };
 
 module.exports.createNewSession = async (state) => {
-  state.id = uuidv4();
+  console.log("createNewSession, state ID:", state.id);
+  // state.id = uuidv4();
```
Comment (dacoaster) [inline src/modules/FileSystemUtility.js:65]: Remove these lines if they are no longer needed.

File: src/modules/FileSystemUtility.js [hunk 2]
```diff
@@ -202,19 +204,18 @@ module.exports.openSession = async () => {
 };
 
 module.exports.exportSession = async (params) => {
-  debugger;
   const timestamp = dayjs().format("YYYY-MM-DD_HH-mm-ss-ms");
-  const id = databaseUtility.getSessionID();
+  // const id = databaseUtility.getSessionID();
```
Comment (dacoaster) [inline src/modules/FileSystemUtility.js:208]: Remove these lines if they are no longer needed.

File: src/store/index.js [hunk 3]
```diff
@@ -174,13 +144,58 @@ export default new Vuex.Store({
...
+  actions: {
+    // async resetState(context, payload) {
+    //   await this._vm.$storageService.updateState(payload);
+    //   context.commit("resetState", payload);
+    // },
+    // async restoreState(context, payload) {
+    //   await this._vm.$storageService.updateState(payload);
+    //   context.commit("restoreState", payload);
+    // },
+  },
```
Comment (dacoaster) [inline src/store/index.js:181]: Remove these lines if they are no longer needed.

File: src/store/modules/config.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      presession: {
+        status: false,
+        tasks: [],
+      },
+      postsession: {
+        status: false,
+        tasks: [],
+      },
+    },
+  }),
```
Comment (dacoaster) [inline src/store/modules/config.js:18]: What about all of the other potential state configs?  Like `hotkeys`, `localOnly`, `theme`, etc?

**Review by dacoaster — COMMENTED** [review]
Thank you for the help here @isntlazy ! I have left comments.

File: package.json [hunk 5]
```diff
@@ -45,6 +45,7 @@
     "ffprobe-static": "^3.1.0",
     "fluent-ffmpeg": "^2.1.2",
     "form-data": "^4.0.0",
+    "json-server": "^0.17.4",
```
[the remark this answers is not in the document]
Comment (isntlazy) [reply package.json:48]: I'm using it for local rest api mocks, I will remove it from here

File: src/components/ControlPanel.vue [hunk 6]
```diff
@@ -721,25 +712,20 @@ import NewSessionDialog from "./dialogs/NewSessionDialog.vue";
 import DurationConfirmDialog from "./dialogs/DurationConfirmDialog.vue";
 import AudioErrorDialog from "./dialogs/AudioErrorDialog.vue";
 import EndSessionDialog from "./dialogs/EndSessionDialog.vue";
-//import MinimizeControlWrapper from "../components/MinimizeControlWrapper.vue";
```
[the remark this answers is not in the document]
Comment (isntlazy) [reply src/components/ControlPanel.vue:724]: sure

File: src/components/ControlPanel.vue [hunk 7]
```diff
@@ -761,7 +747,6 @@ export default {
     DurationConfirmDialog,
     AudioErrorDialog,
     EndSessionDialog,
-    //MinimizeControlWrapper,
```
[the remark this answers is not in the document]
Comment (isntlazy) [reply src/components/ControlPanel.vue:764]: sure

File: src/components/ControlPanel.vue [hunk 8]
```diff
@@ -1136,62 +1021,29 @@ export default {
...
+      }
+      // todo implement web version for this functionality
     },
-    showSourcePickerDialog() {
-      this.fetchSources().then((data) => {
+    async showSourcePickerDialog() {
+      try {
+        let data = await this.fetchSources();
         this.loaded = true;
         this.sources = data.filter((source) => source.name !== "yattie");
```
[the remark this answers is not in the document]
Comment (isntlazy) [reply src/components/ControlPanel.vue:1033]: sure

File: src/components/ControlPanel.vue [hunk 9]
```diff
@@ -1251,35 +1103,34 @@ export default {
...
+        quickTest: this.$store.state.quickTest,
+        path: this.$route.path,
+      };
 
-        window.ipc.invoke(IPC_HANDLERS.FILE_SYSTEM, {
-          func: IPC_FUNCTIONS.CREATE_NEW_SESSION,
-          data: data,
-        });
-      }
+      await this.$storageService.createNewSession(data);
```
[the remark this answers is not in the document]
Comment (isntlazy) [reply src/components/ControlPanel.vue:1129]: I will  check on this, in general I tried to leave the logic as before at this point

File: src/components/MenuPopover.vue [hunk 10]
```diff
@@ -121,33 +113,38 @@ export default {
   },
   methods: {
     async openAccountLink(credentialType, credential) {
+      console.log(credentialType, credential);
```
[the remark this answers is not in the document]
Comment (isntlazy) [reply src/components/MenuPopover.vue:116]: nice catch

File: src/modules/FileSystemUtility.js (hunk 1 above)
Comment (isntlazy) [reply src/modules/FileSystemUtility.js:65]: sure

File: src/modules/FileSystemUtility.js (hunk 2 above)
Comment (isntlazy) [reply src/modules/FileSystemUtility.js:208]: sure

File: src/store/index.js (hunk 3 above)
Comment (isntlazy) [reply src/store/index.js:181]: sure

File: src/store/modules/config.js (hunk 4 above)
Comment (isntlazy) [reply src/store/modules/config.js:18]: yes, will need to add them here for initial state, will do that

**Comment by isntlazy** [comment]
@dacoaster I updated PR with fixes that you suggested and resolves conflicts

File: src/components/ControlPanel.vue [hunk 11]
```diff
@@ -1136,62 +1021,29 @@ export default {
...
+      // todo implement web version for this functionality
     },
-    showSourcePickerDialog() {
-      this.fetchSources().then((data) => {
+    async showSourcePickerDialog() {
+      try {
+        let data = await this.fetchSources();
         this.loaded = true;
         this.sources = data.filter((source) => source.name !== "yattie");
-        if (this.viewMode === "normal") {
```
[the remark this answers is not in the document]
Comment (isntlazy) [reply src/components/ControlPanel.vue:1154]: updated

File: src/components/ControlPanel.vue [hunk 12]
```diff
@@ -1316,61 +1165,26 @@ export default {
...
-        });
-      }
+      this.endSessionDialog = true;
     },
     closeEndSessionDialog(status) {
       this.endSessionDialog = false;
       if (status) {
-        this.summaryDialog = true;
+        this.showSummaryDialog();
+        // this.summaryDialog = true;
```
[the remark this answers is not in the document]
Comment (isntlazy) [reply src/components/ControlPanel.vue]: sure

File: src/components/ControlPanel.vue [hunk 13]
```diff
@@ -1813,52 +1601,24 @@ export default {
...
-        duration: this.duration,
-        sourceId: this.sourceId,
-      };
-      localStorage.setItem("state-data", JSON.stringify(data));
-      if (!window.ipc) return;
-      await window.ipc.invoke(IPC_HANDLERS.WINDOW, {
-        func: IPC_FUNCTIONS.OPEN_MINIMIZE_WINDOW,
-        data: { width: 400, height: 84 },
-      });
-    },
```
[the remark this answers is not in the document]
Comment (isntlazy) [reply src/components/ControlPanel.vue:1829]: sure

**Pull request merged**